nipype/pipeline/plugins/tests/test_base.py

```diff
@@ -39,3 +40,34 @@ def func(arg1):
 
 wf.run(plugin='MultiProc')
 """
+
+@pytest.mark.regression
+def test_remove_nodes(tmpdir):
+    import os
+    import nipype.interfaces.utility as niu
+    import nipype.pipeline.engine as pe
+
+    wf = pe.Workflow(name='test')
+
+    def func(arg1):
+        try:
+            if arg1 == 2:
+                raise Exception('arg cannot be ' + str(arg1))
+        except:
+            pass
+        return arg1
+
+    funkynode = pe.MapNode(niu.Function(function=func, input_names=['arg1'],
+                                        output_names=['out']),
+                           iterfield=['arg1'],
+                           name = 'functor')
+    funkynode.inputs.arg1 = [1,2]
+
+    wf.add_nodes([funkynode])
+    wf.base_dir = tmpdir
+    wf.config['execution']['remove_node_directories'] = 'true'
+    wf.config['execution']['stop_on_first_crash'] = 'false'
+    wf.config['logging']['crashdump_dir'] = tmpdir
+    res = wf.run(plugin='MultiProc')
+
+    assert os.path.isdir(Path(tmpdir)/'test/functor') is False
```

Comment:
You can use instance methods as functions on instances:

```suggestion
    assert not Path.is_dir(tmp_path / "test" / "functor")
```